src/components/Dropdown.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import { Person } from '../types/Person';
+import cn from 'classnames';
+
+type Props = {
+  peopleFromServer: Person[];
+  onSelected: (person: Person | null) => void;
+  delay: number;
+};
+
+export const Dropdown: React.FC<Props> = ({
+  peopleFromServer,
+  onSelected,
+  delay,
+}) => {
+  const [query, setQuery] = useState('');
+  const [isActive, setIsActive] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), []);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+    onSelected(null);
+  };
+
+  const handleSelectPerson = (person: Person) => {
+    setQuery(person.name);
+    onSelected(person);
+    setIsActive(false);
+  };
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(person => {
+      return person.name
+        .toLowerCase()
+        .includes(appliedQuery.toLowerCase().trim());
+    });
+  }, [appliedQuery]);
```

Comment:
Fix all cases please

```suggestion
  }, [appliedQuery, peopleFromServer]);
```

![image](https://github.com/mate-academy/react_autocomplete/assets/70009205/e2b1f94c-1299-428a-a5e4-e8163e9f3357)